.github/workflows/pr.yaml

```diff
@@ -155,3 +157,53 @@ jobs:
 
       - name: Run Javascript tests
         run: dotrun test-js
+
+  publish_report:
+    name: publish-e2e-report
+    # if job is cancelled we should skip this
+    if: success() || needs.e2e.outputs.job_status == 'failure'
+    needs: [e2e]
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    env:
+      HTML_REPORT_URL_PATH: reports/${{ github.head_ref || github.ref_name }}/${{ github.run_id }}/${{ github.run_attempt }}
+    steps:
+      - uses: actions/checkout@main
+        with:
+          ref: gh-pages
+          token: ${{ secrets.GITHUB_TOKEN }}
+      # user git configs are needed for git commands to work
+      # actual authentication is done using secrets.GITHUB_TOKEN with write permission
+      - name: Set Git User
+        run: |
+          git config --global user.email "github-action@example.com"
+          git config --global user.name "GitHub Action"
+      - name: Download zipped HTML report
+        uses: actions/download-artifact@v3
+        with:
+          name: playwright-report
+          path: ${{ env.HTML_REPORT_URL_PATH }}
+      - name: Push HTML Report
+        timeout-minutes: 3
+        # In case if another action job pushed to gh-pages while we are rebasing for the current job
+        run: |
+          git add .
+          git commit -m "workflow: add HTML report for branch ${{ github.head_ref || github.ref_name }} (run_id: ${{ github.run_id }}, attempt:  ${{ github.run_attempt }})"
+
+          while true; do
```

Comment:
Why the loop?

Reply:
same comment as before

Reply:
Makes sense, maybe also add the comment in the line above the loop?

Reply:
sure done